File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/JoustingWeaponTraitEnum.java
package ai.ragu.charactermanager.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.HashMap;
import java.util.Map;

public enum JoustingWeaponTraitEnum implements WeaponTraitEnum {
    JOUSTING_D6(String.format("jousting_%s", DieEnum.D6));

    private final String value;
    private final static Map<String, JoustingWeaponTraitEnum> CONSTANTS = new HashMap<>();

    static {
        for (JoustingWeaponTraitEnum c: values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    JoustingWeaponTraitEnum(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static JoustingWeaponTraitEnum fromValue(String value) {
        JoustingWeaponTraitEnum constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        } else {
            return constant;
        }
    }
}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/JoustingWeaponTraitEnum.kt
package ai.ragu.charactermanager.enumeration

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import java.util.*

enum class JoustingWeaponTraitEnum(private val value: String) : WeaponTraitEnum {
    JOUSTING_D6(String.format("jousting_%s", DieEnum.D6));

    companion object {
        private val CONSTANTS: MutableMap<String, JoustingWeaponTraitEnum> = HashMap()
        @JsonCreator
        fun fromValue(value: String?): JoustingWeaponTraitEnum {
            val constant = CONSTANTS[value]
            return constant ?: throw IllegalArgumentException(value)
        }

        init {
            for (c in values()) {
                CONSTANTS[ai.ragu.charactermanager.enumeration.c.value] = ai.ragu.charactermanager.enumeration.c
            }
        }
    }

    override fun toString(): String {
        return value
    }

    @JsonValue
    fun value(): String {
        return value
    }

}